src/Microsoft.Health.Dicom.Core/Models/Operations/Serialization/OperationTypeConverter.cs

```diff
@@ -0,0 +1,39 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Dicom.Core.Models.Operations.Serialization
+{
+    internal class OperationTypeConverter : JsonConverter<OperationType>
+    {
+        public override bool CanWrite => false;
+
+        public override OperationType ReadJson(JsonReader reader, Type objectType, OperationType existingValue, bool hasExistingValue, JsonSerializer serializer)
+        {
+            if (reader.TokenType is not (JsonToken.String or JsonToken.Null))
+            {
+                throw new JsonReaderException(
+                    string.Format(
+                        CultureInfo.InvariantCulture,
+                        DicomCoreResource.UnexpectedJsonToken,
+                        JsonToken.String,
+                        reader.TokenType));
+            }
+
+            // TODO: Support Nullable<OperationType> is necessary
+            return string.Equals(reader.Value as string, "Reindex", StringComparison.OrdinalIgnoreCase)
```

Comment:
AFAIK, the name of `DurableFunctionStatus` is the function name. so in our case it should be `AddAndReindexTagsAsync`?
I'm thinking should we handle `OperationType` on AzureFunction side instead, since don't want the client know much detail (mapping between functionName and Operationtype).

Reply:
I am unsure about this. I think about it the other way, that the Azure Functions are written agnostic to the structure of how the consumers will transform it. The Azure Functions expose their data in the way that makes the most sense for its domain: the enums, the terminology, the concepts -- at first blush, I don't think our Azure Functions should particularly care about the concepts of the user. Instead, the Dicom Server code, the `IOperationsClient`, acts as an adapter between these two worlds.

However, I can also see a use-case for simply implementing the Azure Functions with these concepts in mind. However, are we limiting ourselves by pushing the public "view" of our long-running jobs so far down the technology stack? My instinct is to do that at the last moment.